src/core/qgsmaplayer.h

```diff
@@ -1489,8 +1499,10 @@ class CORE_EXPORT QgsMapLayer : public QObject
      */
     void clone( QgsMapLayer *layer ) const;
 
-    //! Sets the extent
-    virtual void setExtent( const QgsRectangle &rect );
+    /**
+     * Sets the \a extent in layer CRS.
+     */
+    virtual void setExtent( const QgsRectangle &extent );
```

Comment:
This is technically an api break, since any plugin using 

    layer.setExtent(rect=QgsRectangle(...))

will break